types/address.go

```diff
@@ -709,3 +709,54 @@ func cacheBech32Addr(prefix string, addr []byte, cache *simplelru.LRU, cacheKey
 	}
 	return bech32Addr
 }
+
+type AddressConfig struct {
+	coinType uint32
+	purpose  uint32
+}
+
+var (
+	sdkAddressConfig  *AddressConfig
+	initAddressConfig sync.Once
+)
+
+// GetAddressConfig returns theaddres instance for the SDK.
+func GetAddressConfig() *AddressConfig {
```

Comment:
This is still a global config right? Aren't we just replacing a precious global config by a new one?